### DISCUSSION TITLE
Highlight name of symbol in beginning of Go docstrings

### DISCUSSION THREAD

**Discussion by weakphish** [helix-editor/helix#10472] [Ideas]
In Go, the standard idiom is to begin docstrings with the name of the symbol being described, like so:

```go
// foo performs an action
func foo() {
    fmt.Println("bar")
}

// Bar is a struct
type Bar struct {
    baz int 
}
```
I propose that, similar to Goland from Jetbrains, the first term of a Go docstring be highlighted slightly differently, a bright color perhaps. I'm not sure if this would be a treesitter function or a theme function, but I personally would enjoy it, and thought others may as well.

**Comment by matoous** [comment]
To my understanding, this would have to be addressed in https://github.com/tree-sitter/tree-sitter-go and I don't think there's an easy way to do so as of now. All comments are captured by the same group and ther wouldn't be a way to distinguish docstrings and regular comments. Furthermore, while it is the general practice in Go it is not enforced and/or mandatory so it might not always be the case that the comment starts with the symbol name.

A while back I experimented with doing the same for `go` commands, e.g.

```
//go:generate ./my-custom-tool
```

where `go:generate` would be highlighted and even that was quite challenging with the current tree-sitter-go setup.

**Comment by pascalkuthe** [comment]
If doc documents are separate syntax nodes in go then this could be solved trough an injection parser (similar to how we highlight `TODO` in comments with `tree-sitter-comment`).

**Comment by weakphish** [comment]
@pascalkuthe when you say syntax node, do you mean in tree-sitter? If it is, then I'd be happy to draw up a PR if this is a feature that would be fit for master.
